.github/workflows/pr.yml

```diff
@@ -0,0 +1,63 @@
+name: PR
+
+on:
+  pull_request:
+    branches: [main]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  test-linux:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout source
+      uses: actions/checkout@v2
+
+    - uses: Swatinem/rust-cache@v1
+
+    - name: cargo test
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --workspace
+
+    - name: rustfmt
+      uses: actions-rs/cargo@v1
+      with:
+        command: fmt
+        args: --all -- --check
+
+    - name: clippy
+      uses: actions-rs/clippy-check@v1
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        args: -- -D warnings
```

Comment:
Please add `--tests` here, so that we get lints for tests too

Reply:
What do you think of 
> in order to also check tests and non-default crate features, use 
> `cargo clippy --all-targets --all-features -- -D warnings`

Reply:
That's perfect :)